@xen-orchestra/lite/src/stores/xapi-collection.store.ts

```diff
@@ -123,16 +105,30 @@ const createXapiCollection = <T extends XenApiRecord>(type: RawObjectType) => {
     () => fetchAll()
   );
 
-  const subscribe = () => {
+  function subscribe(
+    options?: SubscribeOptions<true>
+  ): CollectionSubscription<T>;
+
+  function subscribe(
+    options: SubscribeOptions<false>
+  ): DeferredCollectionSubscription<T>;
+
+  function subscribe(
+    options: SubscribeOptions<boolean>
+  ): CollectionSubscription<T> | DeferredCollectionSubscription<T>;
```

Comment:
Yes, it is required.

When you use function overloading, only the overloads are taken into account, not the actual implementation function.

For example, doing this will not work:

```
function foo(arg: string): string; // <= This is a possibility

function foo(arg: number): number; // <= This is another possibility

// This one is only the implementation and thus will not be seen as a possibility
function foo(arg: string | number): string | number {
    return "string" === typeof arg ? "ABC" : 123;
}

function bar(x: string | number) {
    return foo(x);
    //         ^ TypeScript will throw an error here
    //           because there is no foo signature that
    //           can take a string | number.
}```